Internalizes SCT_verify and removes SCT_verify_v1

RJPercival opened the pull request:
SCT_verify is impossible to call through the public API (SCT_CTX_new() is not part of the public API), so this renames it to SCT_CTX_verify and moves it out of the public API.

SCT_verify_v1 is redundant, since SCT_validate does the same verification (by calling SCT_verify) and more. The API is less confusing with a single verification function (SCT_validate).

This also corrects the comment for SCT_verify that omits the fact that it may return values other than 0 and 1.

richsalz commented:
+1 from me, but it's up to @mattcaswell to decide.  This looks like opaque/accessor kinds of things, so we *could* ship it as a bugfix, but probably better to have this in the initial release.

@RJPercival , what manpages do you need help with?  I have some time today.

RJPercival commented:
If you've got the time, it'd be helpful if you could look at how to resolve any unaddressed comments on PR #1372. In the meantime, I'm filling in the remaining PODs.

richsalz commented:
I read through all comments and posted a suggested diff for the one open comment that I found.

RJPercival commented:
Thanks, I've addressed that now.

mattcaswell commented on crypto/ct/ct_locl.h:
```diff
@@ -151,6 +151,13 @@ __owur int SCT_CTX_set1_issuer_pubkey(SCT_CTX *sctx, X509_PUBKEY *pubkey);
+ * Returns 1 if the SCT verifies successfully; any other value indicates
+ * failure. See EVP_DigestVerifyFinal() for the meaning of those values.
+ */
+__owur int SCT_CTX_verify(const SCT_CTX *sctx, const SCT *sct);
```
Strictly speaking we don't usually capitalise internal names (although there are plenty of historic examples where we break that rule). However since the rest of this file also breaks that rule, I guess we let it slide. We can always fix it later as these are purely internal symbols.

mattcaswell commented:
+1

mattcaswell commented:
Merged.
